System/Vector.h

```diff
@@ -17,45 +17,37 @@ namespace RTE {
 
 		SerializableOverrideMethods
 
-		float m_X; //!< X value of this vector.
-		float m_Y; //!< Y value of this vector.
+		float m_X = 0.0F; //!< X value of this vector.
+		float m_Y = 0.0F; //!< Y value of this vector.
 
 #pragma region Creation
 		/// <summary>
-		/// Constructor method used to instantiate a Vector object.
+		/// Constructor method used to instantiate a Vector object with values (0, 0).
 		/// </summary>
-		Vector() { Clear(); }
+		Vector() {};
 
 		/// <summary>
 		/// Constructor method used to instantiate a Vector object from X and Y values.
 		/// </summary>
 		/// <param name="inputX">Float defining the initial X value of this Vector.</param>
 		/// <param name="inputY">Float defining the initial Y value of this Vector.</param>
-		Vector(float inputX, float inputY) { Create(inputX, inputY); }
+		Vector(const float inputX, const float inputY) :
+		m_X(inputX),
+		m_Y(inputY) {};
 
 		/// <summary>
 		/// Copy constructor method used to instantiate a Vector object identical to an already existing one.
 		/// </summary>
 		/// <param name="reference">A Vector object which is passed in by reference.</param>
-		Vector(const Vector &reference) { Create(reference.m_X, reference.m_Y); }
-
-		/// <summary>
-		/// Makes the Vector object ready for use.
-		/// </summary>
-		/// <param name="inputX">Float defining the initial X value of this Vector.</param>
-		/// <param name="inputY">Float defining the initial Y value of this Vector.</param>
-		/// <returns>An error return value signaling success or any particular failure. Anything below 0 is an error signal.</returns>
-		int Create(float inputX, float inputY) { m_X = inputX; m_Y = inputY; return 0; }
+		Vector(const Vector &reference) = default;
 #pragma endregion
 
-#pragma region Destruction
+#pragma region Getters and Setters
 		/// <summary>
 		/// Sets both the X and Y of this Vector to zero.
 		/// </summary>
-		void Reset() { m_X = m_Y = 0.0F; }
-#pragma endregion
+		void Reset() { m_X = 0.0F; m_Y = 0.0F; }
```

Comment:
This is a method that is in usage in cpp, and is also already defined in lua.
Clear() is not defined in lua currently, so this can be a breaking change to scripts that used Reset().
Are you sure?

Reply:
I changed my mind about reset here by the way, I still think it's a little weird in lua context, but not worth making a fuss about. Feel free to ignore and resolve this.